tests/xharness/Jenkins/ErrorKnowledgeBase.cs

```diff
@@ -0,0 +1,69 @@
+using System.IO;
+using Microsoft.DotNet.XHarness.iOS.Shared;
+using Microsoft.DotNet.XHarness.iOS.Shared.Logging;
+
+#nullable enable
+
+namespace Xharness.Jenkins {
+	public class ErrorKnowledgeBase : IErrorKnowledgeBase{
+		static bool IsHE0038Error (ILog log) {
+			if (log == null)
+				return false;
+			if (File.Exists (log.FullPath) && new FileInfo (log.FullPath).Length > 0) {
+				using (var reader = log.GetReader ()) {
+					while (!reader.EndOfStream) {
+						string line = reader.ReadLine ();
+						if (line == null)
+							continue;
+						if (line.Contains ("error HE0038: Failed to launch the app"))
+							return true;
+					}
+				}
+			}
+			return false;
+		}
+
+		static bool IsMonoMulti3Issue (ILog log) {
+			if (log == null)
+				return false;
+			if (File.Exists (log.FullPath) && new FileInfo (log.FullPath).Length > 0) {
+				using var reader = log.GetReader ();
+				while (!reader.EndOfStream) {
+					string line = reader.ReadLine ();
+					if (line == null)
+						continue;
+					if (line.Contains ("error MT5210: Native linking failed, undefined symbol: ___multi3"))
+						return true;
```

Comment:
that's pretty much a dupe of the previous method
and that requires 2 scans (instead of one) for the log files (unless they are different files?)
won't scale well (at least if new ones are copy/pasted)
might be better to give a predicate to a method that scan the provided log

Reply:
atm we only have one, but you are right, is tempting to copy paste and re-parse the file several times.